### PR TITLE
Add command-line arguments as HISTORY to the FITS file

### PR DESCRIPTION
**Pull request by bmerry**
This is made much more complicated by the need to remove
authentication-related information, which precludes just dumping
`sys.argv`. So instead functions are added to reconstruct an equivalent
command line from the parsed args.

Closes SPR1-498.

### CODE REVIEW

**Comment by bmerry** [comment]
I'd suggest starting with arguments.py and test_arguments.py, which should make clear what these new functions do. Everything else is plumbing (and I got sidetracked and made LoaderBase into a proper abstract base class).

File: katsdpimager/frontend.py [hunk 1]
```diff
@@ -279,9 +266,14 @@ def log_parameters(self, suffix=''):
         log_parameters("CLEAN parameters" + suffix, self.clean_p)
 
 
+def prepend_dunder(value):
```
Comment (mauch) [inline katsdpimager/frontend.py:269]: 'dunder' always makes me chuckle:) Strictly speaking should you call a '--' a dunder? I always thought it was '__'.

File: katsdpimager/arguments.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
+"""Utilities for processing and reconstructing command-line arguments."""
+
+import argparse
+from typing import List, Iterable, Set, Mapping, Callable, Any
+
+import astropy.units as u
+
+
+def parse_quantity(str_value):
```
[the remark this answers is not in the document]
Comment (bmerry) [reply katsdpimager/arguments.py:9]: Actually I should delete it: it looks like you can just use the Quantity constructor directly these days!

File: katsdpimager/frontend.py (hunk 1 above)
Comment (bmerry) [reply katsdpimager/frontend.py]: Good point! I've renamed to `prepend_dashes`.

File: katsdpimager/arguments.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    - All arguments must have a default, even if it is ``None``. In other
+      words, do not set the default to ``argparse.SUPPRESS``.
+    - The default must have a meaningful equality comparison with the
+      argument type.
+    - It will break down if the default is a string but the argument type is
+      not a string (in this case, argparse first sets it to the string value,
+      then replaces it with the conversion of that string value).
+
+    Access to the information is provided by :func:`argument_changed` rather
+    than a methods to avoid potential conflicts with command-line option
```
[the remark this answers is not in the document]
Comment (bmerry) [reply katsdpimager/arguments.py]: Actually I meant "than via methods". Fixed.

**Pull request merged**